lib/grammars.coffee

```diff
@@ -134,6 +134,22 @@ module.exports =
       command: "julia"
       args: (context) -> [context.filepath]
 
+  Kotlin:
+    "Selection Based":
+      command: "bash"
+      args: (context) ->
+        code = context.getCode(true)
+        tmpFile = GrammarUtils.createTempFileWithCode(code, ".kt")
+        jarName = context.filename.replace(/\.kt$/g, "") + ".jar"
+        args = ['-c', "kotlinc " + tmpFile + " -include-runtime -d " + jarName + " && java -jar " + jarName + " && rm " + jarName]
```

Comment:
Agreed, it's pretty awesome! Now that I think of it adding vanilla Java support should be about the same (at least for a file based run).